Tests/LeafKitTests/LeafKitTests.swift

```diff
@@ -913,6 +913,54 @@ final class LeafKitTests: XCTestCase {
 
         XCTAssertEqual(view.string, "Hello barvapor")
     }
+	
+	func testCachePurging() throws {
+        var state = false
+        var test = ReplacingFiles(state: &state)
+        
+        test.baseFiles["/foo.leaf"] = """
+        Goodbye
+        """
+        test.baseFiles["/bar.leaf"] = """
+        #extend("foo") cache control
+        """
+        test.newFiles["/foo.leaf"] = """
+        Hello
+        """
+        test.newFiles["/bar.leaf"] = """
+        #extend("foo") purging
+        """
+        test.newFiles["/baz.leaf"] = """
+        #extend("foo") cache control
+        """
+        
+        let renderer = LeafRenderer(
+            configuration: .init(rootDirectory: "/"),
+            cache: DefaultLeafCache(),
+            files: test,
+            eventLoop: EmbeddedEventLoop()
+        )
+        
+        // baseFiles templates are used for reference
+        var view = try renderer.render(path: "bar", context: [:]).wait()
+		XCTAssertEqual(view.string, "Goodbye cache control")
+
+        // newFiles templates are now used for reference, but cache maintains old ones
+        state = true
```

Comment:
I think making `ReplacingFiles` a class would make this test case a lot more readable. Instead of pointers and base / new files, you could just do:

```swift
try renderer.render(...)
testFiles["foo.leaf"] = newFile
try renderer.render(...)
try renderer.cache.remove(...)
try renderer.render(...)
```

Even better would be if we re-use `TestFiles` for this case.

Reply:
Yeah - was avoiding messing with already written code, but it is much cleaner - just switched TestFiles to a class.